harness/determined/keras/_tf_keras_trial.py

```diff
@@ -773,19 +773,41 @@ def _control_loop(self) -> None:
         for wkld, args, response_func in self.workloads:
             logging.debug(f"Received wkld {wkld.kind} with args {args}.")
             if wkld.kind == workload.Workload.Kind.RUN_STEP:
-                # Configure the state for a training step.
-                self.train_response_func = response_func
-                self.train_workload_batches = 0
-                self.train_workload_metrics = []
-                self.train_workload_len = wkld.num_batches
-                self.multiplexer.set_batches_requested(wkld.num_batches)
-                break
+                try:
```

Comment:
There is no need to capture an exception here. You should instead capture the exception in `TFKerasTrialController.run`.